src/panels/mailbox/ha-dialog-show-audio-message.js

```diff
@@ -0,0 +1,129 @@
+import '@polymer/paper-button/paper-button.js';
+import '@polymer/paper-dialog/paper-dialog.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../../resources/ha-style.js';
+
+import LocalizeMixin from '../../mixins/localize-mixin.js';
+
+/*
+ * @appliesMixin LocalizeMixin
+ */
+class HaDialogShowAudioMessage extends LocalizeMixin(PolymerElement) {
+  static get template() {
+    return html`
+    <style include="ha-style-dialog">
+      @media all and (max-width: 500px) {
+        paper-dialog {
+          margin: 0;
+          width: 100%;
+          max-height: calc(100% - 64px);
+
+          position: fixed !important;
+          bottom: 0px;
+          left: 0px;
+          right: 0px;
+          overflow: scroll;
+          border-bottom-left-radius: 0px;
+          border-bottom-right-radius: 0px;
+        }
+      }
+
+      paper-dialog {
+        border-radius: 2px;
+      }
+      paper-dialog p {
+        color: var(--secondary-text-color);
+      }
+
+      #mp3dialog paper-icon-button {
+        float: right;
+      }
+    </style>
+    <paper-dialog id="mp3dialog" with-backdrop opened="{{_opened}}" on-opened-changed="_openedChanged">
+      <h2>
+        [[localize('ui.panel.mailbox.playback_title')]]
+        <paper-icon-button
+          on-click='openDeleteDialog'
+          icon='hass:delete'
+        ></paper-icon-button>
+      </h2>
+      <div id="transcribe"></div>
+      <div>
+        <audio id="mp3" preload="none" controls> <source id="mp3src" src="" type="audio/mpeg" /></audio>
+      </div>
+    </paper-dialog>
+`;
+  }
+
+  static get properties() {
+    return {
+      hass: Object,
+
+      _currentMessage: Object,
+
+      _opened: {
+        type: Boolean,
+        value: false,
+      },
+    };
+  }
+
+  showDialog({ hass, message }) {
+    this.hass = hass;
+    this._currentMessage = message;
+    this.$.mp3.src = null;
+    this._opened = true;
+    this.$.transcribe.innerText = message.message;
+    var platform = message.platform;
+    var mp3 = this.$.mp3;
+    var url = 'mailbox/media/' + platform + '/' + message.sha;
+    // Pass authorization in request:
+    //   https://stackoverflow.com/questions/46878787/
+    //           how-to-set-loaded-audio-to-a-html-audio-tag-controller
+    var oReq = new XMLHttpRequest();
+    var auth = this.hass.connection.options.auth;
+    oReq.open('GET', '/api/' + url, true);
+    oReq.responseType = 'blob';
+    oReq.setRequestHeader('authorization', `Bearer ${auth.accessToken}`);
+    oReq.onload = function () {
+      var blob = oReq.response; // Note: not oReq.responseText
+      if (blob) {
```

Comment:
Should probably show an error if this is not the case. Otherwise it's just an empty `<audio>` element

Reply:
well, the camera stuff is another place where we do binary transfer, and it just uses a passed token rather and disables the Auth code.  So it seems like either we enhance callApi or we pass a token as part of the URL and disable the auth requirement in the mailbox component.  Doing that is also a little simpler in the frontend.  But I'm not sure if the token setup is legacy and not the recommended way forward, or if it would be acceptable.

Reply:
The camera tokens are really only used for the "stream" url when the more-info dialog is open. When we just need a single frame we use the websocket api to fetch the image and set the src directly. I don't think tokens are the correct way to do this, especially now that we have the auth system

Reply:
No I think your solution is probably better. I don't know people are relying on it for custom-ui or anything. My version probably wouldn't be breaking for our code, but your version won't break anything

Reply:
Ok.  I updated the callApi code.  Hopefully this works for everyone